### PR TITLE
TypeScript: Models(Image)

### PR DESCRIPTION
**Pull request by alukach**
## How I did it

---

This PR builds off of #218 and should be reviewed after that is merged.

Part of #187

### CODE REVIEW

File: src/api/db/models/Image.ts [hunk 1]
```diff
@@ -427,25 +462,6 @@ export class ImageModel {
...
-
-      for (let oid = 0; oid < input.objects.length; oid++) {
-        const image = await ImageModel.queryById(input.objects[oid].imageId, context);
-
-        for (let lid = 0; lid < (input.objects[oid].labels || []).length; lid++) {
-          input.objects[oid].labels[lid] = reviewerLabelRecord(
-            project,
-            image,
-            input.objects[oid].labels[lid],
-          );
```
Comment (alukach) [inline src/api/db/models/Image.ts:444]: I believe this was a bug.  It should have been `input.objects[oid].object.labels[lid]` as per the types:

https://github.com/tnc-ca-geo/animl-api/blob/2c66fa1413d35fa6710486019548b64f2364efbc/src/api/type-defs/inputs/CreateObjectsInput.ts#L1-L17

File: src/api/type-defs/inputs/CreateLabelsInput.ts [hunk 2]
```diff
@@ -13,8 +13,9 @@ export default /* GraphQL */ `
     labeledDate: Date
     validation: ValidationInput
     userId: ID
-    imageId: ID
+    imageId: ID!
     objectId: ID
+    mlModel: String
```
Comment (alukach) [inline src/api/type-defs/inputs/CreateLabelsInput.ts:18]: I believe that this was mistakenly omitted from the input. The code that expects this input examines the value for the `mlModel` property:

https://github.com/tnc-ca-geo/animl-api/blob/2c66fa1413d35fa6710486019548b64f2364efbc/src/api/db/models/Image.js#L715-L718

File: src/api/type-defs/inputs/CreateLabelsInput.ts [hunk 3]
```diff
@@ -13,8 +13,9 @@ export default /* GraphQL */ `
     labeledDate: Date
     validation: ValidationInput
     userId: ID
-    imageId: ID
+    imageId: ID!
```
Comment (alukach) [inline src/api/type-defs/inputs/CreateLabelsInput.ts:16]: Required to make this work correctly:

https://github.com/tnc-ca-geo/animl-api/blob/76df33d98667de6bfdf837bb55fec9bbbf0226d9/src/api/db/models/Image.ts#L689-L702

File: src/api/type-defs/inputs/CreateInternalLabelsInput.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ export default /* GraphQL */ `
     bbox: [Float!]!
     mlModel: String!
     mlModelVersion: String!
-    imageId: ID
+    imageId: ID!
```
Comment (alukach) [inline src/api/type-defs/inputs/CreateInternalLabelsInput.ts:8]: Required to make this work correctly:

https://github.com/tnc-ca-geo/animl-api/blob/76df33d98667de6bfdf837bb55fec9bbbf0226d9/src/api/db/models/Image.ts#L560-L578

**Review by nathanielrindlaub — APPROVED** [review]

**Pull request merged**